tendermint/src/lib.rs

```diff
@@ -39,7 +39,7 @@ pub mod node;
 pub mod private_key;
 pub mod public_key;
 pub mod rpc;
-mod serializers;
+pub mod serializers;
```

Comment:
Do we want to export our serializers? IMO it's better to keep it unexported if possible.

Reply:
It's exported so the tests, in a completely different package can access them. I wasn't going to originally, but that's how our testing is arranged.

Reply:
We are basically committing to a public API here (which I don't think we should in this case). We should keep that surface small (as devs using this library will rely on it). 

Either we need to limit the visibility to where it is necessary. Alternatively the tests for a particular type could go as `mod test` into that type (I think that is preferred for simple unit test). See e.g. how `test_serialization_roundtrip` is used in other places.

Reply:
Ok, so testing needs to move within the serialization library. I can refactor here or in a new PR. Which one would you prefer?

Reply:
Actually, since testing changes with the step 2 and step 3 stuff, I'd rather do it as step 4 in a new PR... I just have to make sure I do it before any releases...

Feels crappy to be honest. If we don't want to publicize the serializers, they shouldn't end up on the master branch. Opinion?

Reply:
> Ok, so testing needs to move within the serialization library. I can refactor here or in a new PR. Which one would you prefer?

It doesn't really matter as long as we do it eventually.

> Feels crappy to be honest. If we don't want to publicize the serializers, they shouldn't end up on the master branch

Not sure I understand: I mean that we should make helper-functions used to deserialze actual types not public (only in this crate of where necessary). Why wouldn't these end up on the master branch? There are a bunch of private (or crate public structs/methods on master).

Users of tendermint-rs shouldn't need to care about deserialzation details. We should be careful what API we make public (as it is a commitment, people will end up using it and be annoyed if we break the API). 
And from the previous discussion it sounded the main motivation was the way we structured tests (which isn't a good reason to publish something that previously wasn't public).

Does that make sense?

Reply:
Sorry, I overthought it. I understand that we should move the testing into the serialization library, so we don't have to publish the serialization library.
I was thinking about implementing this here and now: move the testing into the serialization library in this PR. But the next PR already open is based on this and that would also need an overhaul, up until the third PR is merged.

So instead, I propose that we open an issue that tracks this (make it high priority so I implement it before we cut a new release) and do it there.

Reply:
Opened issue #263 to track this.